resolwe/elastic/builder.py

```diff
@@ -10,33 +10,99 @@
 from importlib import import_module
 import inspect
 
+from elasticsearch_dsl.connections import connections
+
 from django.apps import apps
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
 
 from .indices import BaseIndex
 
 
-index_builders = []
+__all__ = ('index_builder',)
+
+
+class ElasticSignal(object):
+    """Class for creating signals to update/create indexes.
+
+    To register the signal, add the following code::
+
+        from django.dispatch import receiver
+
+        signal = ElasticSignal(<my_signal>, <method_name>)
+        receiver(<signal_type>, [sender=<my_model>])(signal)
+
+    ``signal type`` can be i.e. ``django.db.models.signals.pre_save``.
+
+    """
+
+    def __init__(self, index, method_name):
+        """Initialize signal."""
+        self.index = index
+        self.method_name = method_name
+
+    def call(self, sender, instance, **kwargs):
+        """Process signal."""
+        method = getattr(self.index, self.method_name)
+        method(obj=instance)
+
+
+class IndexBuilder(object):
+    """Find indexes and register coresponding signals.
+
+    TODO
+
+    """
+
+    #: list of index builders
+    indexes = []
+
+    def __init__(self):
+        """Initialize index builder object."""
+        # Set dafault connection for ElasticSearch
+        # TODO: Reestablish connection if broken!
```

Comment:
Is this not handled automatically (e.g. retry API request)? Because AFAIK connections are not persistent, each request is done via HTTP?